core/src/main/java/org/apache/carbondata/core/datastore/page/statistics/ColumnPageStatsVO.java

```diff
@@ -96,8 +96,7 @@ public void update(Object value) {
       case DOUBLE:
         max = ((double) max > (double) value) ? max : value;
         min = ((double) min < (double) value) ? min : value;
-        int num = getDecimalCount((double) value);
-        decimal = decimal > num ? decimal : num;
+        decimal = getDecimalCount((double) value);
```

Comment:
I check java.math.BigDecimal.valueOf(7756787654567891L).multiply(java.math.BigDecimal.valueOf(100000000)).doubleValue() is not equal to 775678765456789098765432.789d, but equal to Math.round(775678765456789098765432.789d).

So in such case, I think it is not right to use any Upscale codec then. I think we need to choose the CompressionCodec to compress it only. What do you think?
@kumarvishal09 please check